src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -54,15 +54,16 @@ public class PipeOptions
             {
                 pauseWriterThreshold = DefaultPauseWriterThreshold;
             }
-            else if (pauseWriterThreshold < 0)
-            {
-                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.pauseWriterThreshold);
-            }
 
             if (resumeWriterThreshold == -1)
             {
                 resumeWriterThreshold = DefaultResumeWriterThreshold;
             }
+
+            if (pauseWriterThreshold < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.pauseWriterThreshold);
+            }
```

Comment:
```suggestion
            else if (resumeWriterThreshold == 0)
            {
                // A resumeWriterThreshold of 0 makes no sense because the writer could never resume if paused.
                // By setting it to 1, the writer will resume only after all data is consumed.
                resumeWriterThreshold = 1;
            }
```